lms/djangoapps/discussion/tests/test_views.py

```diff
@@ -188,18 +189,37 @@ def make_mock_perform_request_impl(
     def mock_perform_request_impl(*args, **kwargs):
         url = args[1]
         if url.endswith("threads") or url.endswith("user_profile"):
-            return {
-                "collection": [
-                    make_mock_thread_data(
-                        course=course,
-                        text=text,
-                        thread_id=thread_id,
-                        num_children=None,
-                        group_id=group_id,
-                        commentable_id=commentable_id,
-                    )
+            if is_cohorted:
+                thread_data = [
+                    {"thread_id": thread_id, "group_id": group_id, "commentable_id": commentable_id},
+                    {"thread_id": "global_group_thread_id", "group_id": group_id, "commentable_id": "global_group_discussion"},  # pylint: disable=line-too-long
+                    {"thread_id": "alpha_group_thread_id", "group_id": 1, "commentable_id": "alpha_group_discussion"},
+                    {"thread_id": "beta_group_thread_id", "group_id": 2, "commentable_id": "beta_group_discussion"},
                 ]
-            }
+                return {
+                    "collection": [
+                        make_mock_thread_data(
+                            course=course,
+                            text=text,
+                            num_children=None,
+                            **data
+                        )
+                        for data in thread_data
+                    ]
+                }
+            else:
```

Comment:
I think we can avoid `else` here by placing adjacent code block at level of upper `if`